src/libsyntax/parse/parser.rs

```diff
@@ -233,6 +233,21 @@ macro_rules! maybe_whole {
     )
 }
 
+/// Uses $parse_expr to parse an expression and returns the span of the interpolated
+/// token or the span of the parsed expression, if it was not interpolated
+macro_rules! interpolated_or_expr_span {
```

Comment:
Yeah, that makes sense. I've pushed another commit that turns it into a function.